src/neptune_tensorboard/integration/tensorflow_integration.py

```diff
@@ -208,20 +78,29 @@ def image(name, data, step=None, max_outputs=3, description=None):
         shape = tf.shape(data)
         if len(shape) >= 4:
             for num in range(0, shape[0]):
-                current_step = step + float(num) / (int(shape[0]) + 1)
-                experiment_getter().log_image(
-                    get_channel_name(name), x=current_step, y=data[num], description=description
-                )
+                run[base_namespace]["image"][get_channel_name(name)].append(File.as_image(data[num]))
         else:
-            experiment_getter().log_image(get_channel_name(name), x=step, y=data, description=description)
+            run[base_namespace]["image"][get_channel_name(name)] = File.as_image(data)
         _image(name, data, step, max_outputs, description)
 
     def text(name, data, step=None, description=None):
-        if step is None:
-            step = tf.summary.experimental.get_step()
-        experiment_getter().log_text(get_channel_name(name), x=step, y=data)
+        run[base_namespace]["text"][get_channel_name(name)] = data
         _text(name, data, step, description)
 
+    def graph(graph_data):
+        if IS_GRAPHLIB_AVAILABLE:
+            graph = tfg.board(graph_data)
+            graph.format = "png"
+            fname = str(uuid.uuid4())
+            graph.render(fname)
```

Comment:
How about `graph.pipe(format='pdf')` and using the in-memory files from neptune client? This will dump a file to disk and apply cleanup automatically after uploading a file to Neptune.

Reply:
Makes sense. Thank you!